pylint_plugins/conf_plugin.py

```diff
@@ -0,0 +1,90 @@
+
+import astroid
+
+from astroid import MANAGER
+from pylint import lint
+
+def __init__(self):
+  pass
+
+
+#didn't change/work on this yet
+def wrap_add_message(func):
+  def add_message(msg_descr, line=None, node=None, args=None, confidence=None):
+    kwargs = {'confidence': confidence} if confidence else {}
+    if (msg_descr != 'no-member' or
+        args[0:2] != ('Module', 'openhtf.conf')):
+      return func(msg_descr, line, node, args, **kwargs)
+    # We're sure it's a no-member warning for the openhtf.conf module.
+    try:
+      if args[2] in node.root()._declared_conf_keys:
+        # We found a declaration for the symbol, just skip this message.
+        return
+    except AttributeError:
+      pass
+
+    # TODO(maddychan): Figure out how to add a custom failure message here.
+    return func('no-member', line, node, args, **kwargs)
+
+  return add_message
+
+#storing a reference to the openhtf.conf module node
+clss = []
+clss_locals = {}
+current_root = []
+
+def trans_expr(node):
```

Comment:
a side effect of doing it this way is that i believe it depends on the `conf` module being processed before any that reference it, because otherwise your `clss` and `clss_locals` aren't set.  that might be implied, astroid may parse the modules that a module imports before parsing the module itself.  if you've tried this and it works, then neat.